mdx-models/src/main/java/com/mx/path/model/mdx/model/MdxBase.java

```diff
@@ -5,12 +5,19 @@
 
 import com.mx.path.core.common.model.ModelBase;
 import com.mx.path.core.common.model.Warning;
+import com.mx.path.core.context.Session;
 
 public abstract class MdxBase<T> extends ModelBase<T> {
   private String userId;
 
   private List<Warning> warnings;
 
+  public MdxBase() {
+    if (Session.current() != null) {
+      this.setUserId(Session.current().getUserId());
```

Comment:
If Session exists, userId will always be set, correct?

Reply:
Um... no. 🤔 There may be other problems with this solution.

Reply:
Should we go w/ my solution for a short term fix to unblock the failing endpoints?

Reply:
@gerfboy what if we just also check that `Session.current().getUserId() != null` ?

Then this solution is still viable I believe

Reply:
So, we only setUserId if Session.current().getUserId() isn't blank? That would be an improvement, but are we ok with there just not being a user id in those cases? Also, do we have any timing concerns here? Whereas this is running in the constructor of MdxBase, and the UserProvider version could have been called later, maybe there could be a time where the userId just hasn't made it to the session yet, for some MdxBases.

Sorry, just thinking out loud really.

In general I'm good with that for now so we can keep moving forward, but we should think it through and test it for sure.

Reply:
Looks like we set it initially on AuthenticationsController https://github.com/mxenabled/path-mdx-model/blob/master/mdx-web/src/main/java/com/mx/path/model/mdx/web/controller/AuthenticationController.java#L87

So this solution is viable if we ensure that each connector returns `userId` w/ successful authentication.

I don't think this will matter in the case of Locations because it isn't scoped to user.

@gerfboy does that sound good to you?

Reply:
yeah, I think so.

Reply:
Pushed change that adds check against Session.current().getUserId not being null or ""